Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -348,24 +354,57 @@ private class TopSitesDataSource: NSObject, UICollectionViewDataSource {
     }
 
     private func setDefaultThumbnailBackground(cell: ThumbnailCell) {
-        cell.imageView.image = UIImage(named: "defaultFavicon")!
+        cell.imageView.image = UIImage(named: "defaultTopSiteIcon")!
         cell.imageView.contentMode = UIViewContentMode.Center
     }
 
+    private func getFavicon(cell: ThumbnailCell, site: Site) {
+        // TODO: This won't work well with recycled views. Thankfully, TopSites doesn't really recycle much.'
+        cell.imageView.image = nil
+        cell.backgroundImage.image = nil
+
+        FaviconFetcher.getForUrl(site.url.asURL!, profile: profile) { icons in
+            if (icons.count > 0) {
+                cell.imageView.sd_setImageWithURL(icons[0].url.asURL!) { (img, err, type, url) -> Void in
```

Comment:
Nit: Since we're downloading this url information, could it be possible that it's an invalid url and the force asURL! would crash? Just want to make sure we don't crash if someone feeds us a garbage url.